### PR TITLE
[feature/DBC-25]: api for shops was added

### PR DESCRIPTION
**Pull request by IrinaBarkalova**

### CODE REVIEW

File: retailer/app/delivery/shop/controllers.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from app.delivery.shop.deps import ShopListPagingParams
 from app.delivery.shop.deps import shop_paging_params
 from app.dto.shop import ShopRespDTO, ShopListRespDTO, ShopAddressDTO
+from app.services.shop.shop import ShopsService
```
Comment (ivaaahn) [inline retailer/app/delivery/shop/controllers.py]: в ините добавить импорты по аналогии

File: retailer/app/models/shops.py [hunk 2]
```diff
@@ -15,3 +17,58 @@ class ShopModel(BaseModel):
...
+        return {
+            "id": self.id,
+            "city": self.adr_city,
+            "street": self.adr_street,
+            "house": self.adr_house,
+            "floor": self.adr_floor,
+        }
+
+    @property
+    def adr_city(self) -> str:
```
Comment (ivaaahn) [inline retailer/app/models/shops.py:45]: addreess full

File: retailer/app/repos/shop/implementation.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+from app.base.repo import BasePgRepo
+from app.delivery.shop.deps import shop_paging_params
+from app.dto.shop import ShopListPagingParams
+from app.models.shop_addresses import ShopAddressModel
+from app.models.shops import ShopModel
+
+__all__ = ("ShopsRepo",)
+
+from app.repos.shop.interface import IShopsRepo
```
Comment (ivaaahn) [inline retailer/app/repos/shop/implementation.py]: перед all

File: retailer/app/repos/shop/implementation.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            offset=paging_params.offset,
+            order=paging_params.order,
+            sort=getattr(sp.c, paging_params.sort_by.value),
+        )
+
+        stmt_total = select(func.count()).select_from(sp.join(adr))
+
+        cursor_count = await self._execute(stmt_total)
+        total = cursor_count.scalar()
+        cursor = await self._execute(query)
```
Comment (ivaaahn) [inline retailer/app/repos/shop/implementation.py]: cursor_shops

File: retailer/app/models/shops.py (hunk 2 above)
Comment (IrinaBarkalova) [reply retailer/app/models/shops.py:45]: неть

**Pull request merged**